lib/apollo_tracing/middleware.ex

```diff
@@ -3,52 +3,38 @@ defmodule ApolloTracing.Middleware do
   Documentation for ApolloTracing.
   """
 
-  alias ApolloTracing.{Schema, Schema.Execution, Schema.Execution.Resolver}
+  alias ApolloTracing.Schema.Execution.Resolver
   alias Absinthe.Resolution
 
   # Called before resolving
-  def call(%Resolution{state: :unresolved} = res, _config) do
-    %{acc: %{
-      apollo_tracing_start_time: start_mono_time,
-      apollo_tracing: %Schema{
-        execution: %Execution{
-          resolvers: resolvers_so_far
-        }
-      }
-    }} = res
-
+  # if there isn't an `ApolloTracing` flag set then we aren't actually doing any tracing
+  def call(%Resolution{acc: %{apollo_tracing_start_time: start_mono_time}, state: :unresolved} = res, _config) do
```

Comment:
The pattern match here, and the fall through pattern on line 26 handle situations where the user doesn't actually want to run tracing for a given run.

Reply:
@athal7 This also handles the issue you were facing with the `prime` function.